pennylane/transforms/hamiltonian_expand.py

```diff
@@ -77,6 +101,24 @@ def hamiltonian_expand(tape):
         raise ValueError(
             "Passed tape must end in `qml.expval(H)`, where H is of type `qml.Hamiltonian`"
         )
-
-    hamiltonian.simplify()
-    return qml.transforms.measurement_grouping(tape, hamiltonian.ops, hamiltonian.coeffs)
+    if group:
+        hamiltonian.simplify()
+        return qml.transforms.measurement_grouping(tape, hamiltonian.ops, hamiltonian.coeffs)
+
+    # create tapes that measure the Pauli-words in the Hamiltonian
+    tapes = []
+    for ob in hamiltonian.ops:
+        new_tape = tape.copy()
+        new_tape._measurements = [
+            qml.measure.MeasurementProcess(return_type=qml.operation.Expectation, obs=ob)
+        ]
+        tapes.append(new_tape)
+
+    # create processing function that performs linear recombination
+    def processing_fn(res):
+        dot_products = [
+            qml.math.dot(qml.math.squeeze(res[i]), hamiltonian.coeffs[i]) for i in range(len(res))
+        ]
+        return qml.math.sum(qml.math.stack(dot_products))
```

Comment:
```suggestion
        return qml.math.sum(qml.math.stack(dot_products), axis=0)
```

This change allows it to work with shot batching:

```python
import pennylane as qml
from pennylane import numpy as np

dev = qml.device("default.qubit", wires=2, shots=[(1000, 2)])

@qml.qnode(dev)
def circuit(weights, coeffs):
    qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1])
    H = qml.Hamiltonian(coeffs, obs)
    return qml.expval(H)

obs = [qml.PauliZ(0), qml.PauliX(0) @ qml.PauliZ(1)]
coeffs = np.array([0.1, 0.2], requires_grad=True)
weights = np.random.random([2, 2, 3], requires_grad=True)
```

```pycon
>>> print(circuit(weights, coeffs))
[-0.143  -0.1362]
>>> print(qml.jacobian(circuit, argnum=1)(weights, coeffs))
[[-0.288 -0.614]
 [-0.266 -0.622]]
```

Reply:
(maybe worth adding a test?)

Reply:
Wow! I will do, thanks so much!